### PR TITLE
RAII for SortedBlock

### PR DESCRIPTION
**Pull request by lnkuiper**
This PR moves `block_idx`, `entry_idx` and `BufferHandle` away from `SortedBlock`, and into `SBScanState`. This allows multiple threads to scan the scan the same `SortedBlock`, each using their own `SBScanState`.

Sorry for the delay @hawkfish ! Please let me know if this was what you had in mind, and if anything needs changing.

### CODE REVIEW

File: src/common/sort/sorted_block.cpp [hunk 1]
```diff
@@ -242,18 +179,25 @@ unique_ptr<SortedBlock> SortedBlock::CreateSlice(const idx_t start, const idx_t
...
 	}
 	// And the payload data
-	result->payload_data =
-	    payload_data->CreateSlice(start_block_index, start_entry_index, end_block_index, end_entry_index);
-	D_ASSERT(result->Remaining() == end - start);
+	result->payload_data = payload_data->CreateSlice(start_block_index, end_block_index, end_entry_index);
+	return result;
+}
+
+unique_ptr<SortedBlock> SortedBlock::Copy() {
```
Comment (Mytherin) [inline src/common/sort/sorted_block.cpp:194]: The Copy() method appears to be unused, can it be removed?

File: src/common/sort/sorted_block.cpp [hunk 2]
```diff
@@ -95,6 +58,17 @@ unique_ptr<SortedData> SortedData::CreateSlice(idx_t start_block_index, idx_t st
 	return result;
 }
 
+unique_ptr<SortedData> SortedData::Copy() {
+	auto result = make_unique<SortedData>(type, layout, buffer_manager, state);
```
Comment (Mytherin) [inline src/common/sort/sorted_block.cpp:62]: Same here.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good. One minor comment:

**Comment by lnkuiper** [comment]
Removed the unused function. Should be good to go now

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**